### PR TITLE
Add an example for SearchBar

### PR DESCRIPTION
**Pull request by QuncCccccc**
This PR is to: 
* Update API doc for `SearchBar`.
* Add an example to show how to use a `SearchBar` as the builder of the `SearchAnchor`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/search_anchor.dart [hunk 1]
```diff
@@ -982,6 +983,21 @@ class SearchController extends TextEditingController {
 /// the search bar. Typically only one or two action icons are included.
 /// These actions can represent additional modes of searching (like voice search),
 /// a separate high-level action (such as current location) or an overflow menu.
+///
+/// {@tool dartpad}
+/// This example shows how to use a [SearchBar] as the builder of the [SearchAnchor].
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/search_anchor.dart:988]: SearchBar isn't a SearchAnchorChildBuilder

File: packages/flutter/lib/src/material/search_anchor.dart [hunk 2]
```diff
@@ -970,9 +970,10 @@ class SearchController extends TextEditingController {
 
 /// A Material Design search bar.
 ///
-/// Search bars include a [leading] Search icon, a text input field and optional
-/// [trailing] icons. A search bar is typically used to open a search view.
-/// It is the default trigger for a search view.
+/// Search bars typically consist of a [leading] search icon, a text input field
```
[the remark this answers is not in the document]
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/search_anchor.dart]: This is much better than my version🥲. Just updated. Thanks a lot for the suggestion!

File: packages/flutter/lib/src/material/search_anchor.dart (hunk 1 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/search_anchor.dart]: Ah right, it should be the builder's return value instead of the builder. Just updated.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**